_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+docker push quay.io/<your-quay-username>/wildfly:wildfly-app
+```
+The `-t` command indicates that we are specifying a tag, which in this case is `wildfly-app`. The tag is used to identify different images inside the same repository. Using this image, we can ensure that we are running your local version of WildFly, as opposed to the version of WildFly that is present on OpenShift. 
+
+== Deploying the Application 
+
+Now that we have our WildFly container image, we can use it to deploy our application on it. Navigate to the `elytron-examples/elytron-oidc-client-scope` directory and create a deployment file for your application using this command: 
+```
+mvn package -Pcloud
+```
+Notice we have a profile called `cloud` inside the `pom.xml` file. The `-Pcloud` tag is making use of this profile designed for deploying this application on the cloud. This will create a `simple-webapp-oidc.war` file inside the `target` directory. 
+
+Next, create a new `Dockerfile` inside the `elytron-oidc-client-scope` project with the following commands inside it: 
+```
+FROM quay.io/<your-user-name>/wildfly:wildfly-app
+ADD target/simple-webapp-oidc.war /opt/server/standalone/deployments/
+```
+Here, we are making use of the container image of WildFly we made earlier and copying over the deployment file to the container image so we can deploy it on OpenShift by running the server. Next, build and push a new image of your application using the following commands: 
+```
+docker build -t quay.io/<your-quay-username>/wildfly:latest .
+docker push quay.io/<your-quay-username>/wildfly:latest
+```
+
+=== Creating a helm release 
+
+Lastly, we will be creating a helm release of our local WildFly image and deploying our application on it using a heml chart. Navigate to `charts` directory and open the `helm.yaml` file. The contents of the file will be as follows: 
+```
+image: 
+  name: quay.io/<Username>/wildfly
+build:
+  enabled: false
+deploy:
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: 'https://sso-some-user.apps.sandbox-m4.g2pi.p1.openshiftapps.com'
+    - name: WILDFLY_OVERRIDING_ENV_VARS
+      value: "1"
+    - name: SUBSYSTEM_LOGGING_PATTERN_FORMATTER_COLOR_PATTERN__PATTERN
+      value: "[redhat-openshift] %-5p %s%e%n"
+```
+There are three fields that we have specified: image, build and deploy. 
+
+Using the image tag, we are specifying that we want the pod to use the image of our application that we just built. Notice that we did not specify the tag. This is because when the helm chart uploads, the `latest` tag is appended to it by default, which is why we tagged our application image as that. If you want to use a different tag, then you can edit this by clicking on the pod, click on `Edit Deployment` and changing it under `Image Name`. 
+
+We have also added some environment variable. Notice the `OIDC_PROVIDER_URL`. The value of this will be the url for your RH-SSO pod. So, copy that value and paste it as the value. 
```

Comment:
s/variable/variables

Reply:
Would be good to show how to find the OIDC_PROVIDER_URL.

Reply:
I am referring to the `OIDC_PROVIDER_URL` variable inside the `helm.yaml` file. Should I talk about how it connects to `OIDC_PROVIDER_URL` field inside `oidc.json` file?

Reply:
Here, I just meant that we should describe how to find the URL for the RH SSO instance (either from the OpenShift console or from the CLI using an `oc` command).